pkg/relayer/relayminer_test.go

```diff
@@ -57,3 +60,72 @@ func TestRelayMiner_StartAndStop(t *testing.T) {
 	err = relayminer.Stop(ctx)
 	require.NoError(t, err)
 }
+
+func TestRelayMiner_Ping(t *testing.T) {
```

Comment:
Is this all the test coverage that there is or did I miss something?

I would love to see coverage over a few more scenarios:

1. Multi-supplier relayminer success.
2. Relayminer with multiple endpoint, including a DNS name, ipv4 address and ipv6 address success
    Mocking the DNS name is pretty straightforward with the [go-mockdns](https://github.com/foxcpp/go-mockdns) package.
3. Multi-endpoint partial success/failure on startup (i.e. some endpoints should be unreachable on startup)
4. Multi-endpoint partial success/failure post-startup (i.e. some endpoints should become unreachable after startup)

Consider refactoring this test code into a [test suite](https://github.com/stretchr/testify?tab=readme-ov-file#suite-package) to de-duplicate common code by moving it into methods on the suite and storing case-specific test state on the suite struct (per test case).